### PR TITLE
[webpack] Do not precompile webpack assets

### PR DESCRIPTION
**Pull request by hallelujah**
~When testing, use either webpack --watch or webpack-dev-server~

~TODO: Document it somewhere~

For some reason we cannot use yarn on production.
So the packs are compiled locally and then added to the tree in `public/packs`

So we should keep the WEBPACKER_PRECOMPILE=false in the CI build

### CODE REVIEW

**Comment by hallelujah** [comment]
To be clear on the behaviour: You precompile first then test.
Before we allow compiling on the demand.

Meaning you either need `webpack-dev-server` or `webpack --watch`. Those will do the precompilation for you on fs event change.

**Comment by mikz** [comment]
@hallelujah I'd like it to work as rails.

If you don't precompile, it compiles assets dynamically (if configured to do so).
If you precompile, it uses those precompiled assets.

By precompile, I strictly mean: `rake assets:precompile`.

It should be just possible to clone the repo, run `rake assets:precompile` and run the tests. Or if I don't precompile, to run the webpack-dev-server and let them be automatically compiled on demand.

**Comment by hallelujah** [comment]
@mikz webpack-dev-server does not compile on demand, it precompiles the assets, running a webserver to serve assets.

The default in rails is NOT using webpack-dev-server but only `webpack` through webpacker which will compile on demand the files it encounters in `javascript_pack_tag` in the views

Still in rails you can run `rake assets:precompile` then webpack will see that the files are already compiled so it does not re-compile them.

Using webpack-dev-server is totally different workflow, it will compile everything when started then on file change will recompile the change bundle.

So what behaviour do you want?

**Comment by hallelujah** [comment]
More documentation here:

https://github.com/rails/webpacker#development

File: config/webpacker.yml [hunk 1]
```diff
@@ -41,7 +41,8 @@ development:
 
 test:
   <<: *default
-  compile: true
+  compile: false
```
Comment (didierofrivia) [inline config/webpacker.yml]: So, we never let webpack to compile on the fly for test and we rely on `assets:precompile`. 
And for dev, we create docu explaining that we need webpack-dev-server (`rake webpack:dev`). Would you add the docu here or separate PR?

**Comment by mikz** [comment]
@hallelujah I think it should work as normal rails.

If you have no precompiled assets, they are compiled on demand.
If you precompile, they are just files on the filesystem, no compiling on demand.

That way:
* it is easy to start and just run tests, assets get compiled dynamically
* it is easy to unify behaviour on CI by precompiling
* if you don't precompile, changes are immediately visible in the tests because they are compiled on demand

**Comment by hallelujah** [comment]
1. I agree with all that, that is why we have the `webpacker.ci.yml` and the `webpacker.yml`.
2. I surely have done a premature optimization. Having the `compile: true` is computing file digest everytime it encounters a pack tag:

Lookup of the asset
https://github.com/rails/webpacker/blob/master/lib/webpacker/helper.rb#L15

Compiling
https://github.com/rails/webpacker/blob/master/lib/webpacker/manifest.rb#L22

Check if it is stale
https://github.com/rails/webpacker/blob/master/lib/webpacker/compiler.rb#L20
https://github.com/rails/webpacker/blob/master/lib/webpacker/compiler.rb#L46-L50

The last part can be costly when having many pack tag and watched files so for CI I prefer disabling this part.

So should we close this as having the `webpacker.ci.yml` is good enough?

**Comment by mikz** [comment]
@hallelujah so, we can set compile to true and it would work as I'm describing, just with one config file?
The only downside is there could be a potential performance issue?

Having webpacker.ci.yml is not good enough for me. I can't precompile assets locally and just run the suite. I'd have to also use different webpacker config.  My point is that there should be just one configuration, with properly configured environments.

**Comment by hallelujah** [comment]
Right now if you do not change anything with master branch it just works as you describe unless you use the docker build. But locally it is working exactly as rails default nothing has changed.
The webpacker.ci.yml file is only for CI for performance reason nothing else.

**Comment by hallelujah** [comment]
Changing the title and the description.
The problem with the `ERROR in Entry module not found: Error: Can't resolve 'babel-loader'` was solved in b366e0441106e5f1c126f9bd9a81faceac349138

Removing the webpacker.ci.yml and re-using the `WEBPACKER_PRECOMPILE=false` for the CI build

TL;DR

We can't use yarn for production because of productization limits. Thus we have added this flag `WEBPACKER_PRECOMPILE=false` to make sure when deployed on production it is not trying to compile webpack assets.

Instead we do precompile the packs locally and add them to the tree.

Removing this flag was an error.
